lib/src/dbs/options.rs

```diff
@@ -77,6 +77,16 @@ impl Options {
 		self.db.as_ref().unwrap()
 	}
 
+	/// Get currently selected Level
+	pub fn current_level(&self) -> Level {
+		match (self.ns.as_ref(), self.db.as_ref()) {
+			(Some(_), Some(_)) => Level::Db,
+			(Some(_), None) => Level::Ns,
+			(None, None) => Level::Kv,
+			_ => unreachable!(),
+		}
+	}
+
```

Comment:
Does the return value of this function have any security implications? If not, shouldn't it be documented/renamed to avoid misuse? If so, shouldn't it consider scope and use `self.auth`?

I'm specifically concerned with scope->db privilege escalation and accidentally letting the user access whatever they select.

Reply:
this method is just a helper for doing this:
```
			InfoStatement::User(user, base) => {
				let level = match base {
					// Get the level from the provided user statement
					Some(val) => val.to_level(),
					// If no level is provided, use the current selected level
					None => opt.current_level(),
				};

				// Check if all the necessary options are set for the given level
				opt.needs(level.to_owned())?;
				// Check if the user is allowed to run the statement on the given level
				opt.check(level.to_owned())?;
```

so technically, yeah, you could potentially trick the current session into thinking you are in a level different from that you signed in, but the actual auth check is performed outside.

See that in `current_level`, we don't even return the NS or the DB and it doesn't do any auth related check.